src/darwin/CHIPTool/CHIPTool/View Controllers/QRCode/QRCodeViewController.m

```diff
@@ -173,6 +175,23 @@ - (void)setupUI
     [_resetButton.widthAnchor constraintEqualToConstant:60].active = YES;
     [_resetButton.bottomAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.bottomAnchor constant:-30].active = YES;
     [_resetButton.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor constant:-30].active = YES;
+    
+    // Redirect button
+    _redirectCustomFlowButton = [UIButton new];
+   [_redirectCustomFlowButton setTitle:@"Redirect" forState:UIControlStateNormal];
+    // TODO: Replace resetView with actual redirect action
+   [_redirectCustomFlowButton addTarget:self action:@selector(resetView:) forControlEvents:UIControlEventTouchUpInside];
+    _redirectCustomFlowButton.backgroundColor = UIColor.systemBlueColor;
+    _redirectCustomFlowButton.titleLabel.font = [UIFont systemFontOfSize:17];
+    _redirectCustomFlowButton.titleLabel.textColor = [UIColor whiteColor];
+    _redirectCustomFlowButton.layer.cornerRadius = 5;
+    _redirectCustomFlowButton.clipsToBounds = YES;
+   [self.view addSubview:_redirectCustomFlowButton];
+    
+    _redirectCustomFlowButton.translatesAutoresizingMaskIntoConstraints = false;
+   [_redirectCustomFlowButton.widthAnchor constraintEqualToConstant:100].active = YES;
+   [_redirectCustomFlowButton.bottomAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.bottomAnchor constant:-30].active = YES;
+   [_redirectCustomFlowButton.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor constant:-100].active = YES;
```

Comment:
if we want to add multiple buttons cleanly, I would be for adding a horizontal stack view that would align these properly near each other and would allow for other buttons to be easily added.

See UIStackView * stackViewButtons = [UIStackView new]; in OnOffViewController.m for example